GUIDE.md

```diff
@@ -0,0 +1,302 @@
+    chain.add_handler(handler);
+}
+
+async fn greet(api: Ref<Api>, chat_id: ChatId, user: User) {
+    let method = SendMessage::new(chat_id, format!("Hello, {}", user.first_name));
+    api.execute(method).await.unwrap();
+}
+
+```
+
+### Access
+
+[AccessPredicate](https://tg-rs.github.io/carapax/carapax/access/struct.AccessPredicate.html) allows to protect handlers.
+It takes an [`AccessPolicy`](https://tg-rs.github.io/carapax/carapax/access/trait.AccessPolicy.html).
+Policy has `is_granted` method which takes a `HandlerInput` and returns a future with `bool` output.
+If `true` is returned - access is granted, `false` - forbidden.
+
+[InMemoryAccessPolicy](https://tg-rs.github.io/carapax/carapax/access/struct.InMemoryAccessPolicy.html) is a policy
+which stores [access rules](https://tg-rs.github.io/carapax/carapax/access/struct.AccessRule.html) in memory.
+
+Let's see how it works:
+
+```rust
+use carapax::{
+    access::{AccessExt, AccessPredicate, AccessRule, InMemoryAccessPolicy},
+    types::Update,
+    Chain, Predicate, HandlerResult,
+};
+
+fn setup(chain: &mut Chain) {
+    let policy = InMemoryAccessPolicy::from(vec![AccessRule::allow_user("username")]);
+    let handler = Predicate::new(AccessPredicate::new(policy), protected_handler);
+    // or using AccessExt
+    // let handler = protected_handler.access(policy);
+    chain.add_handler(handler);
+}
+
+async fn protected_handler(update: Update) {
+    log::info!("Got a new update in protected handler: {:?}", update);
+}
+
+```
+
+Since `Chain` implements `Handler` you can also protect a group of handlers or entire bot.
+Naturally, you can implement your own policy in order to store the access rules and/or a list of banned users in a database or some other storage.
+
+Note that you need to enable `access` feature in `Cargo.toml`:
+
+```toml
+carapax = { version = "0.11.0", features = ["access"] }
+```
+
+
+### Ratelimit
+
+Carapax provides a ratelimit support using [governor](https://crates.io/crates/governor) crate.
+
+There are two type of predicates 
+[`DirectRateLimitPredicate`](https://tg-rs.github.io/carapax/carapax/ratelimit/struct.DirectRateLimitPredicate.html)
+and 
+[`KeyedRateLimitPredicate`](https://tg-rs.github.io/carapax/carapax/ratelimit/struct.KeyedRateLimitPredicate.html)
+
+Direct is used when you need to apply ratelimit for all incoming updates.
+Keyed - when you need to limit updates per chat and/or user.
+
+When limit is reached you can either [discard](https://tg-rs.github.io/carapax/carapax/ratelimit/struct.MethodDiscard.html) the updates,
+or [wait](https://tg-rs.github.io/carapax/carapax/ratelimit/struct.MethodWait.html) for the next available time slot.
+
+Every type of predicate can be used [with](https://tg-rs.github.io/carapax/carapax/ratelimit/struct.Jitter.html) or 
+[without](https://tg-rs.github.io/carapax/carapax/ratelimit/struct.NoJitter.html) jitter.
+
+See [example](examples/app/ratelimit.rs) for implementation details.
```

Comment:
I understand `implementation details` as `how it done inside`, so is this better maybe?
```suggestion
See [example](examples/app/ratelimit.rs) for usage details.
```